execution_engine/src/core/engine_state/upgrade.rs

```diff
@@ -245,10 +245,7 @@ where
             }
         }
         // Return true when Entry points have been removed.
-        !contract
-            .entry_points()
-            .keys()
-            .eq(entry_points_to_check.keys())
+        !contract.entry_points().len() == entry_points_to_check.len()
```

Comment:
Small optimization: Moving this comparison before the `for ... in ... { ... }` loop should improve performance. I.e. if lengths are different then it `has_different_entry_points == true`. If they're the same, we have to compare key-for-key.